driver-mapping/src/main/java/com/datastax/driver/mapping/DefaultPropertyMapper.java

```diff
@@ -0,0 +1,542 @@
+            } catch (NoSuchMethodException ignored) {
+            }
+        }
+        return setter;
+    }
+
+    /**
+     * Infers the Cassandra object name corresponding to given the property name.
+     * <p/>
+     * Most users should rely on the implementation provided here.
+     * It relies on annotation values and ultimately resorts to the
+     * {@link NamingStrategy} configured on this mapper.
+     * <p/>
+     * Subclasses may override this method if they need full control
+     * over generating Cassandra object names.
+     *
+     * @param mappedClass The mapped class; this is necessarily a class annotated with
+     *                    either {@link Table @Table} or
+     *                    {@link UDT @UDT}.
+     * @param propertyName The property name; may not be {@code null} nor empty.
+     * @param annotations  The property annotations (found on its field and getter method); may not be {@code null} but can be empty.
+     * @return The inferred Cassandra object name.
+     */
+    protected String inferMappedName(@SuppressWarnings("unused") Class<?> mappedClass, String propertyName, Map<Class<? extends Annotation>, Annotation> annotations) {
+        if (annotations.containsKey(Computed.class)) {
+            String expression = ((Computed) annotations.get(Computed.class)).value();
+            if (expression.isEmpty())
+                throw new IllegalArgumentException(String.format("Property '%s': attribute 'value' of annotation @Computed is mandatory for computed properties", propertyName));
+            return expression;
+        }
+
+        // If a name is explicitly provided with @Column or @Field, use it
+        boolean caseSensitive = false;
+        String mappedName = null;
+        if (annotations.containsKey(Column.class)) {
+            Column column = (Column) annotations.get(Column.class);
+            caseSensitive = column.caseSensitive();
+            if (!column.name().isEmpty())
+                mappedName = column.name();
+        } else if (annotations.containsKey(com.datastax.driver.mapping.annotations.Field.class)) {
+            com.datastax.driver.mapping.annotations.Field udtMappedField =
+                    (com.datastax.driver.mapping.annotations.Field) annotations.get(com.datastax.driver.mapping.annotations.Field.class);
+            caseSensitive = udtMappedField.caseSensitive();
+            if (!udtMappedField.name().isEmpty())
+                mappedName = udtMappedField.name();
+        }
+        if (mappedName != null) {
+            return caseSensitive ? Metadata.quote(mappedName) : mappedName.toLowerCase();
+        }
+
+        // Otherwise delegate to the naming strategy
+        mappedName = namingStrategy.toCassandraName(propertyName);
+        if (mappedName == null || mappedName.isEmpty())
+            throw new IllegalArgumentException(String.format("Property '%s': could not infer mapped name", propertyName));
+        return Metadata.quote(mappedName);
```

Comment:
Yes, escapeId is the solution here. We can't always quote the name because it's compared against the schema metadata coming from Cassandra, where identifiers are only quoted if necessary.
I think we can make handleId public instead of duplicating the logic, it looks like something that might be useful for end users occasionally.